source/lower-bound.ts

```diff
@@ -0,0 +1,21 @@
+
+// Port of lower_bound from http://en.cppreference.com/w/cpp/algorithm/lower_bound
+// Used to compute insertion index to keep queue sorted after insertion
+export default function lowerBound<T>(array: T[], value: T, comp: (a: T, b: T) => number): number {
```

Comment:
Let's give `comp` a better name. I assume it stands for either `compute` or `comparator`.